barman/retention_policies.py

```diff
@@ -170,6 +189,12 @@ def _backup_report(self, source):
         # NOTE: reverse key orders (simulate reverse chronology)
         i = 0
         for bid in sorted(backups.keys(), reverse=True):
+            if backups[bid].is_incremental and len(backups) > 1:
```

Comment:
I added this `len(backups) > 1` because of the `backup_status` method. It takes into account a dict with only one element! If it happens to be an incremental backup, it will be skipped and we will not get the expected result of getting the status of the backup!

Reply:
I'm not sure if I understand your concerns.

That should work fine at a first glance. I mean, this is the code:

```python
    def backup_status(self, backup_id):
        """Report the status of a backup according to the retention policy"""
        source = self.server.available_backups
        if self.context == "BASE":
            return self._backup_report(source)[backup_id]
```

It runs `_backup_report` passing all available backups, and returns a single key. IAC the `_backup_report` is still called with them all.

Reply:
You are correct!